### PR TITLE
[ENH] golden output tests for `MantisForecaster`

### PR DESCRIPTION
**Pull request by geetu040**
Towards #10672

Add regression tests that validate mantis' sktime implementation outputs against outputs from the original/source implementation ([vfeofanov/mantis](https://github.com/vfeofanov/mantis)).

<details>

<summary>
code used to generate the golden outputs for each case
</summary>

```py

"""Generate reference outputs directly from the Mantis source implementation.

This script mirrors exactly what sktime.forecasting.mantis.MantisForecaster
does internally (rolling-window construction, resizing to the model's
``seq_len`` via linear interpolation, embedding extraction, closed-form Ridge
regression on the embeddings, recursive multi-step prediction) but calls
``mantis.architecture`` / ``mantis.trainer.MantisTrainer`` directly, without
importing sktime's forecaster.

Both the embedding extraction (network in ``eval()`` mode, ``no_grad``, no
dropout) and the Ridge fit (closed-form solve) are deterministic given the
input data and model weights, so there is no real source of randomness here.
``torch.manual_seed`` is still set for defensive reproducibility (e.g. in case
of future randomized components), and everything is run on CPU in float32 to
avoid cross-device/dtype nondeterminism.
"""

import numpy as np
import torch
from sklearn.linear_model import Ridge

SEED = 0
FH = 12


def load_airline_values():
    from sktime.datasets import load_airline

    y = load_airline()
    return y.to_numpy(dtype=np.float32).reshape(-1, 1)


def resize_windows(x, seq_len):
    """Resize windows to the model's sequence length, mirroring MantisForecaster._resize_windows."""
    if x.shape[-1] == seq_len:
        return x

    x_tensor = torch.tensor(x, dtype=torch.float32)
    x_resized = torch.nn.functional.interpolate(
        x_tensor, size=seq_len, mode="linear", align_corners=False
    )
    return x_resized.numpy()


def make_tabular_training_data(y, context_length):
    """Replicate MantisForecaster._make_tabular_training_data for a single series."""
    windows, targets = [], []
    for cutoff in range(context_length, len(y)):
        windows.append(y[cutoff - context_length : cutoff].T)
        targets.append(y[cutoff])
    return (
        np.asarray(windows, dtype=np.float32),
        np.asarray(targets, dtype=np.float32),
    )


def load_network(model_version, checkpoint, seq_len, device):
    from mantis.architecture import MantisV1, MantisV2

    if model_version == "v1":
        network = MantisV1(seq_len=seq_len, device=device)
    elif model_version == "v2":
        network = MantisV2(device=device)
    else:
        raise ValueError("model_version must be either 'v1' or 'v2'.")
    return network.from_pretrained(checkpoint)


def run_case(
    model_version, checkpoint, context_length, seq_len, fh=FH, batch_size=256
):
    """Reproduce MantisForecaster.fit(...).predict(...) for a zero-shot Ridge head."""
    from mantis.trainer import MantisTrainer

    torch.manual_seed(SEED)
    device = "cpu"

    network = load_network(model_version, checkpoint, seq_len, device)
    trainer = MantisTrainer(device=device, network=network)

    y = load_airline_values()
    y_train = y[:-fh]

    x_train, y_targets = make_tabular_training_data(y_train, context_length)
    x_train_resized = resize_windows(x_train, seq_len)
    embeddings = trainer.transform(x_train_resized, batch_size=batch_size)

    regressor = Ridge().fit(embeddings, y_targets)

    window = y_train[-context_length:].copy()
    preds = []
    for _ in range(fh):
        x_window = window.T[None, :, :].astype(np.float32)
        x_window_resized = resize_windows(x_window, seq_len)
        embedding = trainer.transform(x_window_resized, batch_size=batch_size)
        next_value = np.asarray(regressor.predict(embedding), dtype=np.float32)
        next_value = next_value.reshape(1, 1)
        preds.append(next_value[0])
        window = np.vstack([window[1:], next_value])

    preds = np.asarray(preds)
    return preds[:3, 0]


if __name__ == "__main__":
    print("=== MantisV2, paris-noah/MantisV2, context_length=64, seq_len=64 (no resize) ===")
    print(
        run_case(
            model_version="v2",
            checkpoint="paris-noah/MantisV2",
            context_length=64,
            seq_len=64,
        )
    )

    print("=== MantisV1, paris-noah/Mantis-8M, context_length=24, seq_len=512 (upsampled) ===")
    print(
        run_case(
            model_version="v1",
            checkpoint="paris-noah/Mantis-8M",
            context_length=24,
            seq_len=512,
        )
    )

```

</details>

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Can you comment why the rtol/atol is big enough to not cause sporadic failures?

Would setting a random seed be an alternative option?

**Comment by geetu040** [comment]
> Would setting a random seed be an alternative option?

I've tried setting seed but it still doesn't give consistent outputs, as a matter of fact, the tags already says `"property:randomness": "stochastic"`, so I think it's rather not a seeding problem.

> Can you comment why the rtol/atol is big enough to not cause sporadic failures?

collecting stats from the last failing CI [runs/30808008777](https://github.com/sktime/sktime/actions/runs/30808008777/job/91668390596)

| OS | Python | Case | Mismatched | Max abs diff | Max rel diff |
|---|---|---|---|---|---|
| ubuntu-latest | 3.10/3.11/3.12/3.13 (identical) | mantisv2-64-64-no-resize | 1/3 | 0.01373291 | 3.1267e-05 |
| ubuntu-latest | 3.10/3.11/3.12/3.13 | mantis8m-v1-24-512-upsampled | 2/3 | 0.00726318 | 1.8471e-05 |
| windows-latest | 3.10/3.11/3.12 (identical) | mantisv2-64-64-no-resize | 1/3 | 0.02072144 | 4.7178e-05 |
| windows-latest | 3.10/3.11/3.12 | mantis8m-v1-24-512-upsampled | 2/3 | 0.00595093 | 1.5133e-05 |
| macos-latest | 3.10 | mantisv2-64-64-no-resize | 1/3 | 0.01620483 | 3.6895e-05 |
| macos-latest | 3.10 | mantis8m-v1-24-512-upsampled | 1/3 | 0.00460815 | 1.1359e-05 |
| macos-latest | 3.11/3.12/3.13 (identical) | mantisv2-64-64-no-resize | 1/3 | 0.01403809 | 3.1962e-05 |
| macos-latest | 3.11/3.12/3.13 | mantis8m-v1-24-512-upsampled | no failure logged (passed) | — | — |

`atol=3e-2` is around 1.5x the largest observed absolute diff (0.0207)
`rtol=1e-4` is around 2x the largest observed relative diff (4.72e-05)
so I think it would probably be fine.

**Comment by fkiraly** [comment]
thanks for reporting - could you assume a half-normal and set the bound at 6 SD, or something similar?

**Comment by geetu040** [comment]
I've left explanation in comments in the commit https://github.com/sktime/sktime/pull/10736/changes/15fabaf82b484eae7cff293b95e7f334c5617212

**Review by fkiraly — APPROVED** [review]
Thanks
